### PR TITLE
feat: add support for reactFlow

### PR DESCRIPTION
**Pull request by arjungarg07**
This PR implements function to generate an array for reactFlow support.

Output Diagram for flight Service.
Screenshot 2021-07-16 at 6 43 36 PM

for PlantUML
Screenshot 2021-07-16 at 6 44 47 PM

### CODE REVIEW

File: lib/getReactFlowData.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/**
+ * Generates reactFlow elements data from default output syntax
+ * 
+ * @param {Object} metrics Relations between AsyncAPI docs in default output syntax
+ * @returns {Array} reactFlow elements data following 
+ */
+
```
Comment (magicmatatjahu) [inline lib/getReactFlowData.js:7]:
```suggestion
```

File: lib/getReactFlowData.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    relations.forEach((operations, channel) => {
+      operations.sub.forEach((metadata,subscriber) => {
+        const service = subscriber.replace(/\s/g,'');
+        if (!visitedServices.includes(service)) {
+          const node = { id: `${service}`, data: { label: `${subscriber}` }, position: { x: 100, y: 10 } };
+          elements.push(node);
+        }
+        const edge = { id: `edge${edgId}`, source: `${service}`, target: `Server${i}`, animated: true, label: `${channel}`, type: 'edgeType', arrowHeadType: 'arrowclosed'};
+        elements.push(edge);
+        edgId++;
```
Comment (magicmatatjahu) [inline lib/getReactFlowData.js:26]: You have duplicated this code blocks two times, so you can move it to the separate function and make configuralne type of operation, for subscribe and publish :)

File: lib/getReactFlowData.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * Generates reactFlow elements data from default output syntax
+ * 
+ * @param {Object} metrics Relations between AsyncAPI docs in default output syntax
+ * @returns {Array} reactFlow elements data following 
+ */
+
+function getReactFlowData(metrics) {
+  let i=1;
+  let edgId = 1;
+  const elements = [];
```
Comment (magicmatatjahu) [inline lib/getReactFlowData.js:11]:
```suggestion
  const nodes = [];
```

Will be better name :)

File: lib/getReactFlowData.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    const node = { id: `Server${i}`, data: { label: `${url}` }, position: { x: 250, y: 5 } };
+    elements.push(node);
+    relations.forEach((operations, channel) => {
+      operations.sub.forEach((metadata,subscriber) => {
+        const service = subscriber.replace(/\s/g,'');
+        if (!visitedServices.includes(service)) {
+          const node = { id: `${service}`, data: { label: `${subscriber}` }, position: { x: 100, y: 10 } };
+          elements.push(node);
+        }
+        const edge = { id: `edge${edgId}`, source: `${service}`, target: `Server${i}`, animated: true, label: `${channel}`, type: 'edgeType', arrowHeadType: 'arrowclosed'};
```
Comment (magicmatatjahu) [inline lib/getReactFlowData.js:24]: We can stay with `animated: true` but I think that we should make it configurable, so user should have possibility to  pass related config object to the given metrics. Could you create issue for that?

File: lib/getReactFlowData.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  const elements = [];
+  const visitedServices = [];
+  metrics.forEach((relations,server) => {
+    const url = server.split(',');
+    const node = { id: `Server${i}`, data: { label: `${url}` }, position: { x: 250, y: 5 } };
+    elements.push(node);
+    relations.forEach((operations, channel) => {
+      operations.sub.forEach((metadata,subscriber) => {
+        const service = subscriber.replace(/\s/g,'');
+        if (!visitedServices.includes(service)) {
```
Comment (magicmatatjahu) [inline lib/getReactFlowData.js:20]: You don't push anything to the `visitedServices`. Is code broken, do you missed about something in implementation? :)

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]
Overall good, but with some comments :) If you have enabled testing framework in project, you should also add tests for it :) Please also add some example (picture) how it looks with react-flow. 

Remember also about Readme.md - you should add doc with new flow :)

File: lib/getReactFlowData.js (hunk 5 above)
Comment (arjungarg07) [reply lib/getReactFlowData.js:20]: aah, I missed it, sorry.

**Comment by arjungarg07** [comment]
Unfortunately, reactFlow have a problem of overlapping edges as we can see in the diagram above, when we compare it to the plantUML diagram it shows that we need flight/update label shown on an edge going from `Flight Monitor Service` to the `server`

**Comment by magicmatatjahu** [comment]
@arjungarg07 There is some blocker for you? I mean, you didn't update PR from 5 days, so I guess that you have holiday, am I right?

**Comment by arjungarg07** [comment]
> @arjungarg07 There is some blocker for you? I mean, you didn't update PR from 5 days, so I guess that you have holiday, am I right?

Yeah, was figuring out the the above mentioned problem.

**Comment by magicmatatjahu** [comment]
@arjungarg07 Reading various similar problems people deal with it this way - https://github.com/wbkd/react-flow/issues/819#issuecomment-758420932 - so then we should implement the custom Edge component. It's related to the PR, you can create issue for that.

**Comment by magicmatatjahu** [comment]
@arjungarg07 If PR will be ready, please make request for the review:

image

**Comment by arjungarg07** [comment]
Sorry, for the delay.

**Comment by magicmatatjahu** [comment]
For me the PR looks good, but I don't know why the coverall throws error. I have to check it out.

**Comment by magicmatatjahu** [comment]
@arjungarg07 I don't have permissions to check error in the https://coveralls.io/, so I will merge PR and will investigate it when Łukasz and Jonas will be back (on the next Monday).

**Review by magicmatatjahu — APPROVED** [review]
LGTM

**Pull request merged**